dulwich/contrib/paramiko_vendor.py

```diff
@@ -111,22 +111,57 @@ def close(self):
 
 
 class ParamikoSSHVendor(object):
-
-    def __init__(self):
-        self.ssh_kwargs = {}
-
-    def run_command(self, host, command, username=None, port=None,
-                    progress_stderr=None):
-        # Paramiko needs an explicit port. None is not valid
-        if port is None:
-            port = 22
+    # http://docs.paramiko.org/en/2.4/api/client.html
+
+    def __init__(self, **kwargs):
+        self.ssh_kwargs = {
+            'port': 22,
```

Comment:
Please don't set defaults here; rather just don't specify them, and paramiko can take care of the defaults.